nntrainer/layers/pooling2d_layer.cpp

```diff
@@ -197,10 +197,21 @@ void Pooling2DLayer::calcDerivative() {
     }
   } break;
   case PoolingType::global_max: {
-    for (unsigned int i = 0; i < deriv.getDim().getDataLen(); ++i) {
-      float der = deriv.getData()[i] / max_idx_global[i].size();
-      for (unsigned int m = 0; m < max_idx_global[i].size(); m++) {
-        result_data[max_idx_global[i][m]] += der;
+    auto iter = max_idx_global.begin();
+    float *deriv_data = deriv.getData();
+    for (unsigned int b = 0; b < batch; ++b) {
+      for (unsigned int c = 0; c < channel; ++c) {
```

Comment:
if some indices are not being used, it would be good to combine them. I am not so sure if the compiler will.

Reply:
Done